src/rpc/util.cpp

```diff
@@ -861,7 +862,7 @@ void RPCResult::ToSections(Sections& sections, const OuterType outer_type, const
 
 bool RPCResult::MatchesType(const UniValue& result) const
 {
-    if (m_skip_type_check) {
+    if (!gArgs.GetBoolArg("-rpcdoccheck", DEFAULT_RPC_DOC_CHECK) || m_skip_type_check) {
```

Comment:
nit: It might be better to put this into `RPCHelpMan::HandleRequest`, so that iteration over the results can be skipped.

Reply:
Just pushed changes. I believe it works but not entirely sure.